models.py

```diff
@@ -104,15 +90,67 @@ class DeckCards(db.Model):
     __tablename__ = "deck_cards"
     id = db.Column(db.Integer, primary_key=True, autoincrement=True)
     deck_id = db.Column(db.Integer, db.ForeignKey("decks.id", ondelete="cascade"))
-    card_id = db.Column(db.Integer, nullable=False)
+    cards_info = db.relationship("Cards")
     card_qty = db.Column(db.Integer, nullable=False)
-    card_name = db.Column(db.Text, nullable=False)
-    card_img = db.Column(db.Text, nullable=False)
-    card_colors = db.Column(db.Text)
-    card_type = db.Column(db.Text, nullable=False)
-    card_cmc = db.Column(db.Integer)
-    card_power = db.Column(db.Integer)
-    card_toughness = db.Column(db.Integer)
+
+
+class CardConnection(db.Model):
+    __tablename__ = "card_connection"
+    card_multiverse_id = db.Column(
+        db.Integer, db.ForeignKey("card_info.card_multiverse_id"), primary_key=True
+    )
+    cards_owned = db.Column(
+        db.Integer, db.ForeignKey("cards_owned.id"), primary_key=True
+    )
+    wish_list = db.Column(
+        db.Integer, db.ForeignKey("card_wish_list.id"), primary_key=True
```

Comment:
I am curious; I notice that this table has four primary keys - one on each column. Typically, when something like this is done, it is called a composite primary key.

To actually create a composite primary key in SQLAlchemy, an extra [step](https://medium.com/@thinesh12/unlocking-the-power-of-composite-primary-keys-in-sqlalchemy-b378fb975e9b#:~:text=Defining%20a%20Composite%20Primary%20Key,within%20your%20SQLAlchemy%20model%20class.) needs to be taken.

I am curious - do these four columns combine together to represent a unique entry? Like no one else can have this combination? I'm assuming so since they all appear to be IDs.

If so, this could be a good use of a composite primary key. It will create an index on these columns, which is an optimization for speeding up reads.

One thing to note about indexes, especially ones that group multiple columns, is that writes can be slower, because the index needs to be rebuilt each time there is a write ( `INSERT`, `UPDATE`, etc.).

However, in this app, the frequency of reads is likely to be much greater than that of writes, so a composite primary key here probably makes sense. It will help if:

1. Your queries use all four keys.
2. Typically, it can also be useful if, when you don't use all keys in a query, you at least use the leading key and maybe one of the others.

All that said, if these do not somehow uniquely identify a record, then it's probably not worth it and no worries.

Reply:
I'm not 100% sure a composite primary key is necessary for this build. The reason I say that is due to loading efficiency. If the index needs to be rebuilt each time there is a write (aka, each time someone adds a card to inventory or their wish list, etc) it will make that process slower than we want it. Also - usually only one to two keys will only be used at one time.